examples/ml-slurm-g4.yaml

```diff
@@ -13,102 +13,322 @@
+            "install_gcsfuse": false,
+            "install_cuda": false,
+            "allow_kernel_upgrades": false,
+            "monitoring_agent": "cloud-ops",
+            "install_managed_lustre": false
+          }
+
+      - type: shell
+        destination: install_slurm.sh
+        content: |
+          #!/bin/bash
+          set -e -o pipefail
+          sudo apt update
+          ansible-pull \
+              -U https://github.com/GoogleCloudPlatform/slurm-gcp -C $(vars.build_slurm_from_git_ref) \
+              -i localhost, --limit localhost --connection=local \
+              -e @/var/tmp/slurm_vars.json \
+              ansible/playbook.yml
+
+- group: image
+  modules:
+  - id: slurm-g4-image
+    source: modules/packer/custom-image
+    kind: packer
+    settings:
+      disk_size: $(vars.image_disk_size_gb)
+      machine_type: $(vars.image_build_machine_type)
+      source_image_family: $(vars.source_image_family)
+      source_image_project_id: [$(vars.source_image_project)]
+      image_family: $(vars.built_image_family)
+      omit_external_ip: false
+
+      metadata:
+        user-data: |
+          #cloud-config
+          create_hostname_file: true
+          write_files:
+          - path: /etc/apt/apt.conf.d/20auto-upgrades
+            permissions: '0644'
+            owner: root
+            content: |
+              APT::Periodic::Update-Package-Lists "0";
+              APT::Periodic::Unattended-Upgrade "0";
+    use:
+    - builder-net
+    - slurm-build-script
+
+- group: cluster-env
+  modules:
+  - id: net0
+    source: modules/network/vpc
+    settings:
+      network_name: $(vars.deployment_name)-net-0
+      subnetworks:
+      - subnet_name: $(vars.deployment_name)-sub-0
+        subnet_region: $(vars.region)
+        subnet_ip: $(vars.net0_range)
+      firewall_rules:
+      - name: $(vars.deployment_name)-internal-0
+        ranges: [$(vars.net0_range)]
+        allow:
+        - protocol: tcp
+        - protocol: udp
+        - protocol: icmp
+
+  # Instructions for Dual NIC on g4-standard-384:
+  # 1. Uncomment the 'net1' module below.
+  # 2. Ensure 'vars.net1_range' is also uncommented and defined.
+  #
+  # - id: net1 # Added second network
+  #   source: modules/network/vpc
+  #   settings:
+  #     network_name: $(vars.deployment_name)-net-1
+  #     subnetworks:
+  #     - subnet_name: $(vars.deployment_name)-sub-1
+  #       subnet_region: $(vars.region)
+  #       subnet_ip: $(vars.net1_range)
+  #     firewall_rules:
+  #     - name: $(vars.deployment_name)-internal-1
+  #       ranges: [$(vars.net1_range)]
+  #       allow:
+  #       - protocol: tcp
+  #       - protocol: udp
+  #       - protocol: icmp
+
+  - id: homefs
+    source: modules/file-system/filestore
+    use:
+    - net0 # Filestore attached to net0
+    settings:
+      filestore_tier: BASIC_SSD
+      size_gb: 2560
+      local_mount: /home
+      reserved_ip_range: $(vars.filestore_ip_range)
```

Comment:
filestore and network should not be part of the cluster deployment group. This allows user to tear down and spin up clusters independently without the need to lose their vpc and filestores